lib/datasource/jenkins/get.ts

```diff
@@ -0,0 +1,102 @@
+import { logger } from '../../logger';
+import { ExternalHostError } from '../../types/errors/external-host-error';
+import { clone } from '../../util/clone';
+import { Http } from '../../util/http';
+import { Release, ReleaseResult } from '../common';
+import { id } from './common';
+
+const http = new Http(id);
+const cacheTimeMin = 60;
+
+const packageInfoUrl =
+  'https://updates.jenkins.io/current/update-center.actual.json';
+
+let lastSync = new Date('2000-01-01');
+let packageReleases: Record<string, ReleaseResult> = Object.create(null);
+
+// Note: use only for tests
+export function resetCache(): void {
+  lastSync = new Date('2000-01-01');
+  packageReleases = Object.create(null);
+}
+
+function processPlugin(name: string, data: Record<string, any>): void {
+  const pluginReleases: Release[] = [data?.previousVersion, data?.version]
```

Comment:
Does the registry only maintain the current version and the previous version?

Reply:
In the current implementation, it only fetches https://updates.jenkins.io/current/update-center.actual.json and there it's only the current and previous one, yes.

There's more data in https://updates.jenkins.io/current/plugin-versions.json but that one lacks plugin metadata like a git repo (which is useful for automatic fetching of release notes)

Reply:
Thanks, I understand now. I think it's important that we use *both* so we can fetch release notes as well as get the full list of versions. For https://updates.jenkins.io/current/update-center.actual.json, we could cache that for a pretty long time (e.g. one day) because source URLs should rarely ever change, and we can always re-query it on demand if there's a *new* plugin in the last <24 hours. Would it be OK with you if we break the datasource out into a separate PR to review and merge first? If so then I'll create a separate issue just for it to discuss the approach.

Reply:
@rarkins yep, sounds good.

Reply:
Please see #6710

Reply:
what happens to this PR now?

Reply:
Keep it as is and we can merge master into it and move it ahead once the datasource PR is merged into master first.

Reply:
Alright, I see. Who should be working on that new issue?

Reply:
Ok, just wanted to make sure we're not stepping on each other toes here.
It will take me some as the scope has increased, so probably that new PR will be on the stall for multiple weeks.

I was wondering if it makes sense to merge the current PR and then to improve it later with the more advanced approach in #6710 to fetch all the releases?

This way it is at least useful for those who want to get upgraded to the latest version (most common use case as this is what usually happens in Jenkins community).

Reply:
I'm worried it could be problematic with Renovate being able to see only the latest and second-latest releases. We don't have datasources like that currently, so it's possible there's an unintended consequence we haven't thought of